conan/tools/cmake/toolchain/blocks.py

```diff
@@ -278,11 +281,16 @@ def context(self):
             raise ConanException('CMakeToolchain needs tools.android:ndk_path configuration defined')
         android_ndk_path = android_ndk_path.replace("\\", "/")
 
+        use_cmake_legacy_toolchain = self._conanfile.conf.get("tools.android:cmake_legacy_toolchain")
```

Comment:
```suggestion
        use_cmake_legacy_toolchain = self._conanfile.conf.get("tools.android:cmake_legacy_toolchain", check_type=bool)
```
Better enforce boolean evaluation?